cluster-autoscaler/hack/update-mods.sh

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -o pipefail
+set -o nounset
+shopt -s lastpipe
+
+err_report() {
+    echo "Exited with error on line $1"
+}
+trap 'err_report $LINENO' ERR
+
+SCRIPT_NAME=$(basename "$0")
+K8S_FORK="kubernetes/kubernetes"
+# Update here when we need a new version of Kubernetes
+K8S_REV="ef7808fec57284582c7bf3fc834570e5769df83e"
+TARGET_MODULE=${TARGET_MODULE:-k8s.io/autoscaler/cluster-autoscaler}
+PRINT_HELP="false"
+UPDATE_GO_MOD="true"
+
+# Packages that are allowed to drift from k8s/k8s
+PERMITTED_PACKAGE_DIFFS=(
+    github.com/aws/aws-sdk-go
+    github.com/stretchr/testify
+)
+
+# Kubernetes staging repos: will need to be periodically updated
+# or update this script to check what is in k8s.io/kubernetes/stagin/src each time
+K8S_STAGING_MODULES=(
+    k8s.io/api
+    k8s.io/apiextensions-apiserver
+    k8s.io/apimachinery
+    k8s.io/apiserver
+    k8s.io/cli-runtime
+    k8s.io/client-go
+    k8s.io/cloud-provider
+    k8s.io/cluster-bootstrap
+    k8s.io/code-generator
+    k8s.io/component-base
+    k8s.io/cri-api
+    k8s.io/csi-translation-lib
+    k8s.io/kube-aggregator
+    k8s.io/kube-controller-manager
+    k8s.io/kube-proxy
+    k8s.io/kube-scheduler
+    k8s.io/kubectl
+    k8s.io/kubelet
+    k8s.io/legacy-cloud-providers
+    k8s.io/metrics
+    k8s.io/node-api
+    k8s.io/sample-apiserver
+    k8s.io/sample-cli-plugin
+    k8s.io/sample-controller
+)
+
+################################################################################
+### Flag and directory validation ##############################################
+################################################################################
+
+if [[ $(basename $(pwd)) != "cluster-autoscaler" ]];then
+  echo "The script must be run in cluster-autoscaler directory"
+  exit 1
+fi
+AUTOSCALE_DIR=$(pwd)
+
+if ! which jq > /dev/null; then
+  echo "This script requires jq command to be available"
+  exit 1
+fi
+
+WORK_DIR="${WORK_DIR:-$(mktemp -d /tmp/ca-update-vendor.XXXX)}"
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -h | --help ) PRINT_HELP="true"; break;;
+        -f | --k8sfork ) K8S_FORK="$2"; shift; shift ;;
+        -r | --k8srev ) K8S_REV="$2"; shift; shift ;;
```

Comment:
Please allow tagname/branchname here. 
Resolve with git rev-parse.

If possible also add an option to automatically commit the changes (that will allow to have consistent commit message for go.mod updates).

Reply:
Since this is just referencing a GitHub raw URL any tag, branch, or commit ref should work fine here

Reply:
Ref: https://github.com/golang/go/issues/35878